packages/big-design/src/components/ButtonGroup/styled.tsx

```diff
@@ -0,0 +1,31 @@
+import { theme as defaultTheme } from '@bigcommerce/big-design-theme';
+import styled, { css } from 'styled-components';
+
+import { ButtonProps } from '../Button';
+import { StyleableButton } from '../Button/private';
+
+export const StyledButton = styled(StyleableButton)<ButtonProps & { isFirst?: boolean; isLast?: boolean }>`
+  border-radius: ${({ theme }) => theme.borderRadius.none};
+  margin-right: -1px;
+
+  ${({ isFirst, theme }) =>
+    isFirst &&
+    css`
+      border-bottom-left-radius: ${theme.borderRadius.normal};
+      border-top-left-radius: ${theme.borderRadius.normal};
+    `}
+
+  ${({ isLast, theme }) =>
+    isLast &&
+    css`
+      border-bottom-right-radius: ${theme.borderRadius.normal};
+      border-top-right-radius: ${theme.borderRadius.normal};
+      margin-right: 0;
+    `}
+    
+  &:focus {
+    z-index: 1;
+  }
```

Comment:
Curious as to why this is needed?

Reply:
the explanation is pretty simple, I want to achieve correct `box-shadow` behavior, pls see the right border on the example below

**Without `z-index`**
<img width="346" alt="Screenshot 2021-04-22 at 10 29 58" src="https://user-images.githubusercontent.com/9980803/115674046-f00f1500-a355-11eb-97a2-b7a6445d0514.png">

**With `z-index`**
<img width="296" alt="Screenshot 2021-04-22 at 10 29 46" src="https://user-images.githubusercontent.com/9980803/115674093-fb624080-a355-11eb-9a38-2fc8ed6eaedc.png">